### PR TITLE
Use `raise .. from ..` to explicitly chain exceptions

### LINKED ISSUE #3679: Fix exception chaining
I recently went over [PyTorch](https://github.com/pytorch/pytorch/pull/43836) and [Detectron2](https://github.com/facebookresearch/detectron2/pull/1945), suggesting a fix in the way that Python 3's exception chaining is used.

As described in detail in [this article](https://blog.ram.rachum.com/post/621791438475296768/improving-python-exception-chaining-with), exception chaining ([PEP 3134](https://www.python.org/dev/peps/pep-3134/)) can be used to make exceptions more user-friendly, and in that case, the syntax `raise new_exception from old_exception` needs to be used.

When `raise .. from ..` is used, there will be a line saying `The above exception was the direct cause of the following exception` between tracebacks. However, when implicitly chaining exceptions (meaning when `raise .. from ..` is not used), the message will be `During handling of the above exception, another exception occurred` which can confuse users.

Specifically, the following should be used in order to chain exceptions:
```python
try:
    something_which_raises_OldError
except OldError as e:
    raise NewError("A more user-friendly exception message.") from e 
```
instead of:
```python
try:
    something_which_raises_OldError
except OldError:
    raise NewError("A more user-friendly exception message.")
```

One example which needs to be fixed is:
https://github.com/PyTorchLightning/pytorch-lightning/blob/3d76f604bdb97d66dd1248540ea088fb18c06a03/pytorch_lightning/utilities/parsing.py#L159-L162

If this suggestion sounds good and reasonable, I'd be happy to create a PR! Let me know your thoughts on this!

### PR DESCRIPTION
**Pull request by akihironitta**
### Motivation
Fixes and closes #3679.

### Description of the changes
I changed lines where `raise`ing an exception in `except` to either:
1. `raise .. from e` to explicitly chain exceptions where the exception `e` has valuable information and seems useful for users.
2. ~`raise .. from None` to unchain exceptions where the previous exception has no valuable information and seems noisy and redundant.~ <- There were no places where this option is applicable in my opinion.

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by akihironitta** [comment]
I see CI failing but I believe these errors are caused not due to the changes I made.

**Review by Borda — APPROVED** [review]
LGTM

File: pytorch_lightning/metrics/metric.py [hunk 1]
```diff
@@ -165,15 +165,15 @@ def aggregate(self, *tensors: torch.Tensor) -> torch.Tensor:
         """
         try:
             return torch.cat(tensors).mean(0)
-        except (ValueError, TypeError):
+        except (ValueError, TypeError) as e:
```
Comment (SkafteNicki) [inline pytorch_lightning/metrics/metric.py:168]:
```suggestion
        except (ValueError, TypeError) as exp:
```

File: pytorch_lightning/metrics/metric.py [hunk 2]
```diff
@@ -165,15 +165,15 @@ def aggregate(self, *tensors: torch.Tensor) -> torch.Tensor:
...
             if isinstance(tensors[0], Mapping):
                 return {k: torch.stack([tensor[k] for tensor in tensors]).mean(0) for k in tensors[0].keys()}
             elif isinstance(tensors[0], Sequence) and not isinstance(tensors[0], torch.Tensor):
                 return tuple([torch.stack(tmp).mean(0) for tmp in zip(*tensors)])
             elif isinstance(tensors[0], torch.Tensor):
                 return torch.stack(tensors).mean(0)
             else:
-                raise TypeError("unknown metric value format to aggregate")
+                raise TypeError("unknown metric value format to aggregate") from e
```
Comment (SkafteNicki) [inline pytorch_lightning/metrics/metric.py:176]:
```suggestion
                raise TypeError("unknown metric value format to aggregate") from exp
```

**Review by SkafteNicki — APPROVED** [review]
LGTM, just two suggestions to be consistent with the naming

File: pytorch_lightning/metrics/metric.py (hunk 2 above)
Comment (akihironitta) [reply pytorch_lightning/metrics/metric.py]: Good catch!

**Comment by rohitgr7** [comment]
@akihironitta have you checked if there are no more such places where this needs to be updated?

**Comment by akihironitta** [comment]
@rohitgr7 Yes, I believe. When exceptions are `raise`d in `except`, they are very likely to try to make their previous exceptions more user-friendly or descriptive, so I listed and checked all the lines where exceptions are `raise`d in `except` clauses with [this script](https://gist.github.com/akihironitta/4223c1b32404b36c1b349d70c4c93b4d).

Below is the list of all lines containing `raise` in `except`:
```
tests/base/models.py#L15
pytorch_lightning/metrics/metric.py#L176
pytorch_lightning/accelerators/ddp2_backend.py#L52
pytorch_lightning/trainer/logging.py#L148
pytorch_lightning/tuner/batch_size_scaling.py#L185
pytorch_lightning/tuner/batch_size_scaling.py#L231
pytorch_lightning/utilities/parsing.py#L162
```

**Review by rohitgr7 — APPROVED** [review]
Nice!

**Review by awaelchli — APPROVED** [review]
